Add integration tests

stefanbuck opened the pull request:
This is a first implementation to load the real OctoLinker browser extension in a Chrome instance using puppeteer.

I've created a separate repository which contains all the test files https://github.com/OctoLinker/e2e. End to end tests are generated on-the-fly based on the repo content so no additional PR in here is required to increase our coverage. 

Before `yarn e2e` invokes the actual tests, it downloads https://github.com/OctoLinker/e2e/blob/master/resource.json which is used to generate our e2e tests. For more details visit https://github.com/OctoLinker/e2e

stefanbuck commented on jest-puppeteer.config.js:
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  launch: {
+    dumpio: true,
+    headless: false,
```
Puppeteer headless doesn't support extensions, right now.
 
 https://github.com/GoogleChrome/puppeteer/issues/2819#issuecomment-401432184

stefanbuck commented:
Hi @josephfrazier 👋 Friendly reminder in case you missed my review request. I'm excited to hear feedback on this

josephfrazier commented:
Thanks for the reminder, I had forgotten about this one. This looks really good overall! I'm excited to see proper Chrome tests, and I've been looking for a reason to get familiar with `puppeteer`.

I do have one question though: Why put the e2e tests in a separate repo? It seems like that makes it harder to keep track of them, versus just keeping them here.

stefanbuck commented:
No worries, happens to me all the time 😉 Puppeteer is really amazing and defiantly something you should look into. But back to topic: 

That's a good question. Initially I thought it would be good just for the sake of separating. Also I was thinking about cases were we have to replicate some special file tree structure. However, right now there is no need for this  and maybe never will be. The only last thing I can think of is when it comes to test functionality related to issues or PullRequests. Such dummy issues / PR I would see in this repo.

Conclusion, I'll merge https://github.com/OctoLinker/e2e into this and we will use the `e2e` repo to test other GitHub related features.

stefanbuck commented:
I deleted that E2E repository and merged all the files into `e2e/fixtures/`. On a PullRequest Puppeteer runs the test against the fork so we can verify that fixtures files are working before we merge the PR. I also added a README to explain how it works. Please free to go wild with commenting typos and other suggestions you have to make this useful for others.

stefanbuck commented:
Yes, I really put a lot of thinking into it, while I had a couple days off 😉 What would you snapshots then? We don't store any information in the DOM, so I don't know what you want to snapshots. 

Once #277 is resolved, we can think about snapshotting the DOM. Although we have to be really carefully what we snapshot to keep those small as possible.

stefanbuck commented:
@josephfrazier I'm going to merge this PR now. Let's keep eye on snapshot testing once #277 is complete. Are you able to review the copy in e2e/readme.md raise a fix PR if needed? That would be really awesome.

josephfrazier commented:
Sounds good, sorry I didn't respond earlier about the snapshotting. I was thinking we can use snapshots to store the expected test results, instead of using the `"//": "..."` "comments". Jest snapshots aren't restricted to (V)DOM objects, so it should be doable. Maybe I'll try to set it up if I find the time.

I'll take a look at the E2E readme now and suggest changes.
